src/android/io/branch/BranchSDK.java

```diff
@@ -763,49 +758,49 @@ public SessionListener(CallbackContext callbackContext) {
             this._callbackContext = callbackContext;
         }
 
-        public SessionListener() {
-            this._callbackContext = null;
-        }
-
         //Listener that implements BranchReferralInitListener for initSession
         @Override
         public void onInitFinished(JSONObject referringParams, BranchError error) {
 
             Log.d(LCAT, "SessionListener onInitFinished()");
 
-            String out = "";
+            String out;
 
-            if (this._callbackContext == null) {
-                return;
-            }
 
             if (error == null) {
 
-                out = String.format("DeepLinkHandler(%s)", referringParams.toString());
 
                 // params are the deep linked params associated with the link that the user clicked -> was re-directed to this app
                 //  params will be empty if no data found.
                 if (referringParams == null) {
                     Log.d(LCAT, "return is null");
-                    return;
                 } else {
                     Log.d(LCAT, "return is not null");
                     Log.d(LCAT, referringParams.toString());
+	            out = String.format("DeepLinkHandler(%s)", referringParams.toString());
+                    webView.sendJavascript(out);
+                }
+
+                if (this._callbackContext != null) {
+                    this._callbackContext.success(referringParams);
                 }
-                    
-                this._callbackContext.success(referringParams);
 
             } else {
                 String errorMessage = error.getMessage();
 
-                out = String.format("NonBranchLinkHandler(%s)", error.toString());
-
                 Log.d(LCAT, errorMessage);
 
-                this._callbackContext.error(errorMessage);
-            }
+		// HURDLR: note that this return value is inconsistent with iOS
+		// iOS returns the original url and a dictionary as follows:
+		// { Unable to process URL", @"error", urlString, @"url" }
+		// Therefore, NonBranchLinkHandler is unreliable
+                out = String.format("NonBranchLinkHandler(%s)", error.toString());
+                webView.sendJavascript(out);
 
-            webView.sendJavascript(out);
+                if (this._callbackContext != null) {
+                    this._callbackContext.error(errorMessage);
+                }
```

Comment:
Historically, the order of operations has been call the global DeepLinkHandler method first, then resolve promises. I restored that order to preserve backwards compatibility.

Also, I removed the return between 776 & 777 because if it existed, then you wouldn't necessarily resolve your promise if you had no parameters.